src/event.rs

```diff
@@ -1,4 +1,18 @@
-//! This module contains the `WindowEvent` type and re-exports glutin event
+//! This module contains the `Event` enum and re-exports glutin event
 //! types.
 
-pub use winit::{ElementState, Event, KeyboardInput, MouseButton, VirtualKeyCode, WindowEvent};
+pub use winit::{ElementState, Event as WinitEvent, KeyboardInput, MouseButton, VirtualKeyCode,
+                WindowEvent};
+
+/// Events that the engine might send to the user.
+pub enum Event {
+    /// An event from the Winit crate
+    WinitEvent(WinitEvent),
```

Comment:
Should we make this implementation detail part of the public API?

Reply:
We already have actually.  Our events thus far have come from Winit.  Now the next logical question is should we leave it that way?  I think so personally, as reimplementing and converting the events could be a pain and cost us cycles unnecessarily.  Winit is also a cross platform library that should work for every use case we have thus far.

Reply:
I still see some issues. E.g. the hierarchy is deeper.

Reply:
Yeah the hierarchy is getting kind of excessive at this point.  Perhaps I can see what I can do about that.